src/datasets/arrow_dataset.py

```diff
@@ -3097,6 +3097,21 @@ def add_item(self, item: dict, new_fingerprint: str):
             fingerprint=new_fingerprint,
         )
 
+    def align_labels_with_mapping(self, label2id: Dict, label_column: str = "labels") -> "Dataset":
+        features = self.features.copy()
+        int2str_function = features[label_column].int2str
+        # Some label mappings use uppercase label names so we lowercase them
+        label2id = {k.lower(): v for k, v in label2id.items()}
+        label_names = list(label2id.keys())
+        features[label_column] = ClassLabel(num_classes=len(label_names), names=label_names)
```

Comment:
Here you use the lowercase label names. IMO we should keep the original ones (before lowering)

Reply:
good idea! done :)